tests/command/cursor-batchsize-002.phpt

```diff
@@ -0,0 +1,84 @@
+--TEST--
+MongoDB\Driver\Command batchSize of zero ignored
+--SKIPIF--
+<?php require __DIR__ . "/../utils/basic-skipif.inc"; ?>
+<?php NEEDS_CRYPTO(); ?>
+<?php NEEDS('STANDALONE'); CLEANUP(STANDALONE); ?>
+--FILE--
+<?php
+require_once __DIR__ . "/../utils/basic.inc";
+
+class Test implements MongoDB\Driver\Monitoring\CommandSubscriber
+{
+    public function executeCommand()
+    {
+        MongoDB\Driver\Monitoring\addSubscriber($this);
+
+        $manager = new MongoDB\Driver\Manager(STANDALONE);
+
+		$bulkWrite = new MongoDB\Driver\BulkWrite;
+
+		for ($i = 0; $i < 5; $i++) {
+		    $bulkWrite->insert(array('_id' => $i));
+		}
+
+		$writeResult = $manager->executeBulkWrite(NS, $bulkWrite);
+		printf("Inserted: %d\n", $writeResult->getInsertedCount());
+
+		$command = new MongoDB\Driver\Command(array(
+		    'aggregate' => COLLECTION_NAME,
+		    'pipeline' => array(
+		        array('$match' => new stdClass),
+		    ),
+		    'cursor' => array('batchSize' => 0)
+		));
+		$cursor = $manager->executeCommand(DATABASE_NAME, $command);
+
+		$cursor->toArray();
+        
+        MongoDB\Driver\Monitoring\removeSubscriber($this);
+    }
+
+    public function commandStarted(MongoDB\Driver\Monitoring\CommandStartedEvent $event)
+    {
+    	$command = $event->getCommand();
+
+    	if ($event->getCommandName() === 'aggregate') {
+            printf("aggregate command specifies batchSize: %d\n", $command->cursor->batchSize);
+        }
+
+        if ($event->getCommandName() === 'getMore') {
+            printf("getMore command specifies batchSize: %s\n", isset($command->batchSize) ? 'yes' : 'no');
+        }
+    }
+
+    public function commandSucceeded(MongoDB\Driver\Monitoring\CommandSucceededEvent $event)
+    {
+    	$reply = $event->getReply();
+
+        if ($event->getCommandName() === 'aggregate') {
+            printf("aggregate response contains %d document(s)\n", count($reply->cursor->firstBatch));
+        }
+
+        if ($event->getCommandName() === 'getMore') {
+            printf("getMore response contains %d document(s)\n", count($reply->cursor->nextBatch));
+        }
+    }
+
+    public function commandFailed(MongoDB\Driver\Monitoring\CommandFailedEvent $event)
+    {
+    }
+}
+
+(new Test)->executeCommand();
+
+?>
+===DONE===
+<?php exit(0); ?>
+--EXPECT--
+Inserted: 5
+aggregate command specifies batchSize: 0
+aggregate response contains 0 document(s)
+getMore command specifies batchSize: no
+getMore response contains 5 document(s)
```

Comment:
According to the [manual](https://docs.mongodb.com/manual/reference/method/db.collection.aggregate/index.html#example-aggregate-method-initial-batch-size), a batchSize of 0 means an empty first batch and is useful for quickly returning a cursor or failure message without doing significant server-side work so the aggregate should return 0 documents and the getMore should return the remaining documents (5 in this case).

Reply:
Ah, that's right. I was only thinking about the two `getMore` requests in my last comment.

Side note: this is related to [PHPLIB-312](https://jira.mongodb.org/browse/PHPLIB-312).